File: 0572-subtree-of-another-tree/0572-subtree-of-another-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    bool identicaltrees(TreeNode* p, TreeNode* q) {
        if (p == NULL && q == NULL) {
            return true;
        }
        if (p == NULL || q == NULL) {
            return false;
        }
        return (p->val == q->val && identicaltrees(p->left, q->left) &&
                identicaltrees(p->right, q->right));
    }
    bool isSubtree(TreeNode* root, TreeNode* subRoot) {
        if (root == NULL && subRoot == NULL) {
            return true;
        }
        if (root == NULL || subRoot == NULL) {
            return false;
        }
        return (identicaltrees(root, subRoot) ||
                isSubtree(root->left, subRoot) ||

                isSubtree(root->right, subRoot));
    }
};

File: 0572-subtree-of-another-tree/0572-subtree-of-another-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {TreeNode} subRoot
 * @return {boolean}
 */
function  isSameTree (p, q) {
    if (p == null && q == null) {
        return true;
    }
    if (p == null || q == null) {
        return false;
    }
    return (p.val == q.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right))

};
var isSubtree = function (root, subRoot) {
    

    if (root == null && subRoot == null) {
        return true
    }

    if (root == null || subRoot == null) {
        return false
    }


    if (isSameTree(root, subRoot)==true) {
        return true;
    }
    if (isSubtree(root.left, subRoot) || isSubtree(root.right, subRoot)) {
        return true
    }
    return false

};